src/main/java/seedu/address/storage/JsonAdaptedPerson.java

```diff
@@ -52,11 +53,13 @@ public JsonAdaptedPerson(Person source) {
         name = source.getName().fullName;
         phone = source.getPhone().value;
 
-        if (source instanceof Patient) {
-            age = ((Patient) source).getAge().value;
+        if (source instanceof Doctor) {
+            age = "";
         } else if (source instanceof Donor) {
             age = ((Donor) source).getAge().value;
-        } else { //TODO: change to else if instanceof Doctor
+        } else if (source instanceof Patient) {
+            age = ((Patient) source).getAge().value;
+        } else {
```

Comment:
do we need else? since its low chance we have another type?

Reply:
Do you think switch case would be better?
then default case would be age = ""; since your last else is also the same

Reply:
Tbh the only reason else exists is cause if not the compiler complains that age is possibly undefined.

Reply:
I think switch case cant be used to check instance of?

Reply:
I think we should leave this section of code as-is, or refactor in another PR. Its actually quite weird that JsonAdaptedPerson expects an age attribute when not all persons will have age

Reply:
Then it becomes unclear that there is a block for doctor class. If we wanna change the implementation of doctor class in the future, we might miss this part

Reply:
> I think we should leave this section of code as-is, or refactor in another PR. Its actually quite weird that JsonAdaptedPerson expects an age attribute when not all persons will have age

Regarding this, I tried to create a class called JsonAdaptedPatient but failed. I figured that for now, we could use the JsonAdaptedPerson class to form `Person` objects, then differentiate the Person objects into Patient, Donor and Doctor